core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts

```diff
@@ -206,20 +205,45 @@ export class DotEmaComponent implements OnInit, OnDestroy {
         data
     }: {
         origin: string;
-        data: { action: CUSTOMER_ACTIONS; payload: DotCMSContentlet };
+        data: { action: CUSTOMER_ACTIONS; payload: DotCMSContentlet | AddContentletPayload };
     }): () => void {
         const action = origin !== this.host ? CUSTOMER_ACTIONS.NOOP : data.action;
 
-        return {
+        return (<Record<CUSTOMER_ACTIONS, () => void>>{
             [CUSTOMER_ACTIONS.EDIT_CONTENTLET]: () => {
+                const payload = <DotCMSContentlet>data.payload;
+
                 this.store.initEditIframeDialog({
-                    inode: data.payload.inode,
-                    title: data.payload.title
+                    inode: payload.inode,
+                    title: payload.title
+                });
+            },
+            [CUSTOMER_ACTIONS.ADD_CONTENTLET]: () => {
+                const payload = <AddContentletPayload>data.payload;
+
+                this.store.initAddIframeDialog({
```

Comment:
```suggestion
                this.store.initActionAdd({
```

Even tho we have an iframe in a dialog right now, it might not be like that in the future.

Reply:
Same for `initEditIframeDialog`